packages/next-auth/src/core/lib/cookie.ts

```diff
@@ -132,7 +132,10 @@ export class SessionStore {
 
     for (const name in req.cookies) {
```

Comment:
@Schniz & @balazsorban44 
I think this needs to be a for-of loop for Map to work.
I'm running into this issue where the cookie isn't being read in the middleware
The for-in loop just iterates the properties of the Map object and not the actual constants